spec/models/configuration_script_base_spec.rb

```diff
@@ -0,0 +1,13 @@
+describe ConfigurationScriptBase do
+  it "has many authentications and vice versa" do
```

Comment:
btw To answer your original question more clearly:

> Do we care about the expected relationship being changed unintentionally by someone touching other part of the code base? If we do, what is the best way to guard against that?"

1) Yes
2) If that inflection were to somehow be deleted, the majority of all the tests you write actually testing this feature will completely explode, as nothing will be associated correctly. That'll be a good indication that someone's unintentionally changed the code :)